06.evaluate-model.py

```diff
@@ -0,0 +1,81 @@
+import keras
+import pickle
+import argparse
+import numpy as np
+import pandas as pd
+from keras.models import load_model
+from keras.preprocessing import sequence
+from sklearn.metrics import log_loss, roc_auc_score, confusion_matrix, classification_report
+from sklearn.metrics import roc_curve, auc
+
+parser = argparse.ArgumentParser()
+parser.add_argument("-l", "--ngrams", type=int, default=3)
+parser.add_argument("-m", "--model_name", default="LSTM")
+parser.add_argument("-n", "--namefile", default="data/annotated_names.tsv")
+args = parser.parse_args()
+
+# Load model
+model = load_model("models/%s.h5" % args.model_name)
+filename = "models/" + args.model_name + "_idx_dic.pkl"
+idx_dic = pickle.load(open(filename, "rb"))
+categories = list(pd.read_csv("models/%s_categories.txt" % args.model_name, header=None, names=['category']).category)
+X_train = np.load("models/%s_X_train.npy" % args.model_name, allow_pickle=True)
+y_train = np.load("models/%s_y_train.npy" % args.model_name, allow_pickle=True)
+X_test = np.load("models/%s_X_test.npy" % args.model_name, allow_pickle=True)
+y_test = np.load("models/%s_y_test.npy" % args.model_name, allow_pickle=True)
+
+
+num_words = len(idx_dic.keys())+1
+feature_len = 20
+batch_size = 128
+
+
+X_train = sequence.pad_sequences(X_train, maxlen=feature_len)
+X_test = sequence.pad_sequences(X_test, maxlen=feature_len)
+print('X_train shape:', X_train.shape)
+print('X_test shape:', X_test.shape)
+
+num_classes = np.max(y_train) + 1
+print(num_classes, 'classes')
+
+print('Convert class vector to binary class matrix '
+      '(for use with categorical_crossentropy)')
+y_train = keras.utils.to_categorical(y_train, num_classes)
+y_test = keras.utils.to_categorical(y_test, num_classes)
+print('y_train shape:', y_train.shape)
+print('y_test shape:', y_test.shape)
+
+
+# Get PubMed author data from iscb-diversity repo
+sdf = pd.read_csv(args.namefile, sep="\t", low_memory=False)
+sdf.dropna(subset=['name'], inplace=True)
+
+sdf['name'] = sdf.name.str.title()
+
+sdf.groupby('ethnicity').agg({'name': 'count'})
+
+
+y_pred = model.predict_classes(X_test, verbose=2)
+y_prob = model.predict_proba(X_test, verbose=2)  # to predict probability
+target_names = list(sdf.ethnicity.astype('category').cat.categories)
+print("Log loss", log_loss(y_test, y_prob))
+
+y_true = np.argmax(y_test, axis=1)
+print("ROC/AUC", roc_auc_score(y_true, y_prob, multi_class='ovo'))
```

Comment:
Does not have to be in this PR, but it would be cool to see this for your ethnicolr model and ethnicolor data. While a different problem, the values are at least comparable in terms of how well each does at their respective problem.

Does this output value get stored anywhere currently?

Reply:
I can make this for ethnicolr, no trouble, but do we want that to be pushed to master if we're not going to address it in the body of the paper?

Doesn't get stored anywhere currently. Want me to change that? We could tack it onto the existing LSTM_auroc.tsv

Reply:
> I can make this for ethnicolr, no trouble, but do we want that to be pushed to master if we're not going to address it in the body of the paper?

Don't you have the code to fit the ethnicolr model in master? If so, then I think it makes sense to include this value, regardless of whether we mention it in the manuscript. If ethnicolr analysis is not part of master, then ignore this.

> Doesn't get stored anywhere currently. Want me to change that? We could tack it onto the existing LSTM_auroc.tsv

Good idea. Adding an extra row for "multiclass_ovo" or "combined" or whatever we call it.